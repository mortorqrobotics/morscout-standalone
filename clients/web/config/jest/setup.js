global.fetch = require("jest-fetch-mock");

const Chance = require("chance");

const chances = {};

const mockMath = Object.create(Math);
mockMath.random = (seed = 42) => {
  chances[seed] = chances[seed] || new Chance(seed);
  const chance = chances[seed];
  return chance.random();
};

global.Math = mockMath;
