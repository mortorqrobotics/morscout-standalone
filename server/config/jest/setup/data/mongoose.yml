Season:
  - _id: 5d7be5a7994e35048b9e5e0e
    year: 2019
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
  - _id: 4d7be5a7994e35048b9e5e0e
    year: 2015
    name: Destination Deep Space
    buildStart: 2019-01-04 12:00:00 -5
    buildEnd: 2019-01-06 12:00:00 -5
    seasonStart: 2019-01-08 12:00:00 -5
    seasonEnd: 2019-02-03 12:00:00 -5
Regional:
  - _id: 6d7be5a7994e35048b9e5e0e
    name: Hi!!!
    key: abc
    url: google.com
    eventType: regional
    startDate: 2019-01-08 12:00:00 -5
    endDate: 2019-02-03 12:00:00 -5
    season: 5d7be5a7994e35048b9e5e0e
    location:
      type: Point
      coordinates: [0,0]
Alliance:
  - _id: 2d7ce5a7994e35048b9e5e0e
    teams:
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
      - 1d7ce5a7994e35048b9e5e0e
Match:
  - _id: 7d7be5a7994e35048b9e5e0e
    redAlliance: 2d7ce5a7994e35048b9e5e0e
    blueAlliance: 2d7ce5a7994e35048b9e5e0e
    winner: Red
    startDate: 2019-02-03 12:00:00 -5
    endDate: 2019-02-03 12:10:00 -5
    Regional: 6d7be5a7994e35048b9e5e0e
User:
  - _id: 8d7be5a7994e35048b9e5e0f
    team: 9d7be5a7994e35048b9e5e0e
    phone: 1334567890
    email: bo@bob.bob
    firstname: Bob
    lastname: Boby
    password: Secret
    username: bob2
Team:
  - _id: 9d7be5a7994e35048b9e5e0e
    regional: 6d7be5a7994e35048b9e5e0e
    number: 1515
    name: MorTorq
    location:
      type: Point
      coordinates: [0,0]
