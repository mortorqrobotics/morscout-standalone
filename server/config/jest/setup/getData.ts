/* eslint-disable security/detect-object-injection */
import { join } from "path";
import { readFile } from "fs-extra";
import { parse } from "yaml";
import * as models from "../../../src/models";

export default async function getData() {
  let data = {};
  try {
    data = parse(
      await readFile(join(__dirname, "data", "mongoose.yml")).toString()
    );
  } catch (e) {
    console.log(e);
  }
  let promises: Promise<any>[] = [];
  Object.entries(models).forEach(([name, model]) => {
    if (typeof data[name] !== "undefined") {
      data[name].forEach(datum => {
        promises.push((new model(datum) as any).save());
        // (m as any).save().catch(() => {});
      });
    }
  });
  await Promise.all(promises);
}
