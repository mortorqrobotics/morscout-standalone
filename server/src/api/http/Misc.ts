import getSeason from "api/misc/getSeason";
import { Router, Request, Response } from "express";

export default (router: Router) => {
  router.get("/season", async (req: Request, res: Response) => {
    res.end(JSON.stringify(await getSeason()));
  });
  router.get("/season/:year", async (req: Request, res: Response) => {
    const yearString = req.params.yearString || "2015";
    if (!/[\d]*/.test(yearString)) res.status(400).end("Year must be a number");
    else {
      const year = parseInt(yearString);
      if (year < 2015 || year > new Date().getFullYear())
        res.status(400).end("Year must be between 2015 and now");
      else res.end(JSON.stringify(await getSeason(year)));
    }
  });
};
