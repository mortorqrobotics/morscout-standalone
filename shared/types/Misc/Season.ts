export const loadSeason = "GET_SEASON";
export const updatedSeason = "UPDATED_SEASON";
