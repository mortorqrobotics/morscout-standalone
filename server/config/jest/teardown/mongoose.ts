import mongoose from "mongoose";

module.exports = () => {
  mongoose.disconnect();
  (mongoose as any).server.stop();
};
