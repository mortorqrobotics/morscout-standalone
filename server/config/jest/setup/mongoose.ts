import mongoose from "mongoose";
import { MongoMemoryServer } from "mongodb-memory-server";
import getData from "./getData";

let mongoServer: MongoMemoryServer;
export async function setup() {
  mongoServer = new MongoMemoryServer();
  const mongoUri = await mongoServer.getConnectionString();
  (process as any).mongoUri = mongoUri;
  await mongoose.connect(
    mongoUri,
    { useNewUrlParser: true, useCreateIndex: true },
    err => {
      if (err) console.error(err);
    }
  );
  await getData();
  return mongoUri;
}

export async function teardown() {
  mongoServer.stop();
}
