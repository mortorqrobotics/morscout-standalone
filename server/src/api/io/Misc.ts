import { loadSeason, updatedSeason } from "shared/types/misc/Season";
import getSeason from "api/misc/getSeason";
import { Socket } from "socket.io";

export default (socket: Socket) => {
  socket.on(loadSeason, async (year: number) => {
    socket.emit("action", {
      type: updatedSeason,
      data: await getSeason(year)
    });
  });
};
