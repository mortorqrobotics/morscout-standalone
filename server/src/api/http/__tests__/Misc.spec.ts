import { setup, teardown } from "setup/mongoose";
import express from "express";
import Misc from "../Misc";
import request from "supertest";

beforeAll(setup);
afterAll(teardown);

let router: express.Router;

beforeEach(() => {
  router = express.Router();
  jest.spyOn(router, "get");
  Misc(router);
});

test("miscelaneous routes are the same", async () => {
  expect(
    (router.get as jest.Mock).mock.calls.map(val => val[0]).sort()
  ).toMatchSnapshot();
});

test("/season returns the current season", async () => {
  const req = await request(router).get("/season");
  expect(req.status).toBe(200);
  expect(req.text).toMatchSnapshot();
});

test("/season/year returns the correct year", async () => {
  const req = await request(router).get("/season/2015");
  expect(req.status).toBe(200);
  expect(req.text).toMatchSnapshot();
});
