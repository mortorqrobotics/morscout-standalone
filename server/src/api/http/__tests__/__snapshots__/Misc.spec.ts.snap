// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/season returns the current season 1`] = `"{\\"_id\\":\\"5d7be5a7994e35048b9e5e0e\\",\\"year\\":2019,\\"name\\":\\"Destination Deep Space\\",\\"buildStart\\":\\"2019-01-04T17:00:00.000Z\\",\\"buildEnd\\":\\"2019-01-06T17:00:00.000Z\\",\\"seasonStart\\":\\"2019-01-08T17:00:00.000Z\\",\\"seasonEnd\\":\\"2019-02-03T17:00:00.000Z\\",\\"__v\\":0}"`;

exports[`/season/year returns the correct year 1`] = `"{\\"_id\\":\\"4d7be5a7994e35048b9e5e0e\\",\\"year\\":2015,\\"name\\":\\"Destination Deep Space\\",\\"buildStart\\":\\"2019-01-04T17:00:00.000Z\\",\\"buildEnd\\":\\"2019-01-06T17:00:00.000Z\\",\\"seasonStart\\":\\"2019-01-08T17:00:00.000Z\\",\\"seasonEnd\\":\\"2019-02-03T17:00:00.000Z\\",\\"__v\\":0}"`;

exports[`miscelaneous routes are the same 1`] = `
Array [
  "/season",
  "/season/:year",
]
`;
